### PR TITLE
Support Jenkins Folders in JenkinsArtifactDownload task

### PR DESCRIPTION
**Pull request by kasubram**

### CODE REVIEW

File: Tasks/JenkinsDownloadArtifacts/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -121,5 +121,6 @@
   "loc.messages.UnableToFetchItem": "Unable to fetch item: {0}. Error: {1}",
   "loc.messages.UploadingItem": "Uploading {0}.",
   "loc.messages.FoundJobType": "Found job type %s",
-  "loc.messages.CannotFindJobType": "Could not detect job type"
+  "loc.messages.CannotFindJobType": "Could not detect job type",
+  "loc.messages.InvalidJobName": "Invalid Job name %s"
```
Comment (GitHubSriramB) [inline Tasks/JenkinsDownloadArtifacts/Strings/resources.resjson/en-US/resources.resjson]: Invalid job name %s

File: Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts [hunk 2]
```diff
@@ -70,12 +70,12 @@ async function doWork() {
             }
         }
         else {
-            const artifactQueryUrl: string = `${serverEndpointUrl}/job/${jobName}${jenkinsJobDetails.multiBranchPipelineUrlInfix}/${jenkinsJobDetails.buildId}/api/json?tree=artifacts[*]`;
+            const artifactQueryUrl: string = `${serverEndpointUrl}/${jenkinsJobDetails.jobUrlInfix}/${jenkinsJobDetails.multiBranchPipelineUrlInfix}/${jenkinsJobDetails.buildId}/api/json?tree=artifacts[*]`;
```
Comment (GitHubSriramB) [inline Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts]: assuming you have validated multi-branch pipeline within folder scenario?

File: Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts [hunk 3]
```diff
@@ -26,6 +32,19 @@ export class JenkinsJobDetails {
         this.isMultiBranchPipeline = this.jobType.toLowerCase() === JenkinsJobTypes.MultiBranchPipeline.toLowerCase();
         this.multiBranchPipelineUrlInfix = this.isMultiBranchPipeline ? `/job/${this.multiBranchPipelineName}` : "";
     }
+
+    public static GetJobUrlInfix(jenkinsJobName: string): string {
```
Comment (GitHubSriramB) [inline Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts]: add l0 test for this method?

File: Tasks/JenkinsDownloadArtifacts/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (kasubram) [reply Tasks/JenkinsDownloadArtifacts/Strings/resources.resjson/en-US/resources.resjson]: Fixed.

File: Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts (hunk 2 above)
Comment (kasubram) [reply Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts]: yes, tested.

File: Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts (hunk 3 above)
Comment (kasubram) [reply Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts]: done

**Pull request merged**